### DISCUSSION TITLE
Evaluate new Swift APIs

### DISCUSSION THREAD

**Discussion by feralbob** [outfoxx/Shield#66] [Ideas]
Might be worth evaluating how these new libraries fit into Shield (or indeed how they don't)

https://swift.org/blog/swift-certificates-and-asn1/

**Comment by kdubb** [comment]
They're on my radar as well. The biggest reason for Shield was our projects need the ability to generate X.509 certificates and PKCS#10 certification requests. `swift-certificates` is well on the way to fulfilling that need for certificates and plans on supporting certification requests.

Given these goals, their prominence and backing by Apple, I would happily move `Shield` to incorporate `swift-certificates` once it fulfills our needs. Alternatively, maybe it becomes obsolete.

**Comment by kdubb** [comment]
Maybe obsolescence is a bit much because we are wrapping `Security` in a pretty nice way for multi-platform usage.

**Comment by GeoffreyCzajkowskiK** [comment]
Hey there feralbob,

Thanks for sharing the link to the new Swift libraries for certificates and ASN.1. It's always exciting to see new APIs being released, especially when they can potentially improve our workflow and efficiency.
